v2/internal/genericarmclient/useragent_policy.go

```diff
@@ -41,3 +40,8 @@ func (p userAgentPolicy) Do(req *policy.Request) (*http.Response, error) {
 	req.Raw().Header.Set("user-Agent", newUserAgent)
 	return req.Next()
 }
+
+// AddToUserAgent appends the given suffix to the default user agent.
+func AddToUserAgent(suffix string) {
+	userAgent += " " + suffix
```

Comment:
Suggest only adding the suffix if the value passed in is non-empty. As it stands, if nothing is configured, we add an empty space to the end of the UA.

Reply:
Fixed, thanks!